tools/bpf2c/bpf_code_generator.cpp

```diff
@@ -111,6 +112,25 @@ static std::map<uint8_t, std::string> _opcode_name_strings = {
     ADD_OPCODE(EBPF_OP_JSLT_REG),   ADD_OPCODE(EBPF_OP_JSLE_IMM),  ADD_OPCODE(EBPF_OP_JSLE_REG),
 };
 
+static const std::string _map_type_names[]{
```

Comment:
Since this array must be kept up to date as new entries are added to the bpf_map_type enum, I would recommend adding a comment above the typedef in ebpf_struct.h saying to also update the table here if any are added in the future.  Same for a comment above the ebpf_pin_type enum typedef.

For evidence of why such a comment is important, see the _map_type_names[] array at the top of libs/ebpfnetsh/maps.cpp which is already out of date, so I think that file should also be mentioned in ebpf_struct.h.  (And there's one in external/bpftool/src/map.c as well)